libs/langchain-community/src/stores/message/dynamodb.ts

```
@@ -117,80 +158,129 @@ export class DynamoDBChatMessageHistory extends BaseListChatMessageHistory {
+    }
   }
 
   /**
    * Adds a new message to the DynamoDB table.
    * @param message The message to be added to the DynamoDB table.
    */
   async addMessage(message: BaseMessage) {
-    const messages = mapChatMessagesToStoredMessages([message]);
+    try {
+      const messages = mapChatMessagesToStoredMessages([message]);
 
-    const params: UpdateItemCommandInput = {
-      TableName: this.tableName,
-      Key: this.dynamoKey,
-      ExpressionAttributeNames: {
-        "#m": this.messageAttributeName,
-      },
-      ExpressionAttributeValues: {
-        ":empty_list": {
-          L: [],
+      const params: UpdateItemCommandInput = {
+        TableName: this.tableName,
+        Key: this.dynamoKey,
+        ExpressionAttributeNames: {
+          "#m": this.messageAttributeName,
         },
-        ":m": {
-          L: messages.map((message) => {
-            const dynamoSerializedMessage: DynamoDBSerializedChatMessage = {
-              M: {
-                type: {
-                  S: message.type,
-                },
-                text: {
-                  S: message.data.content,
-                },
-              },
-            };
-            if (message.data.role) {
-              dynamoSerializedMessage.M.role = { S: message.data.role };
-            }
-            return dynamoSerializedMessage;
-          }),
+        ExpressionAttributeValues: {
+          ":empty_list": {
+            L: [],
+          },
+          ":m": {
+            L: messages.map(this.createDynamoDBSerializedChatMessage),
+          },
         },
-      },
-      UpdateExpression:
-        "SET #m = list_append(if_not_exists(#m, :empty_list), :m)",
-    };
-    await this.client.send(new UpdateItemCommand(params));
+        UpdateExpression:
+          "SET #m = list_append(if_not_exists(#m, :empty_list), :m)",
+      };
+      await this.client.send(new UpdateItemCommand(params));
+    } catch (error) {
+      console.error("Error adding message:", error);
```

Same as above, let's not hide the error